### PR TITLE
#167729792-ft-one-way-trip-request

### PR DESCRIPTION
**Pull request by okabamac**
#### What does this PR do?
This PR enables user to request a one-way trip

#### Description of Task to be completed?
- create controller to handle one-way
- create methods to switch request
- validate user inputs
- write test for controller
- ensure appropriate response items are returned

#### How should this be manually tested?
 - clone repository
- startup terminal
- cd into the cloned repository
- start the server by typing ```npm run dev``` into the terminal
- visit via postman and make a POST request with 'one-way' in the request body to
 ```http://localhost:3000/api/v1/trips/requests```
Using the following data as an example 
```
{
"currentOfficeLocation": 1,
"destination": 2,
"accommodation": 2,
"reason": "holiday",
"departureDate": "2019-09-30"
} 
```
#### Any background context you want to provide?
- User must be signed up and logged in
- User must a valid, unexpired token.

#### What are the relevant pivotal tracker stories?
[#167729792](https://www.pivotaltracker.com/story/show/167729792)

#### Screenshots (if appropriate)
Screenshot from 2019-09-11 21-44-31
Screenshot from 2019-09-13 09-56-24

#### Questions:
None

### CODE REVIEW

File: src/controllers/Trips.js [hunk 1]
```diff
@@ -2,28 +2,156 @@
...
+      }
+      const {
+        userId,
+        linemanager: lineManagerUser,
+        firstname: firstName,
+        lastname: lastName,
+        notifyemail: notifyEmail
+      } = aUser;
+      if (!lineManagerUser) {
+        return serverResponse(res, 400, ...['error', 'message', 'Line Manager must be present to continue']);
```
Comment (sebaztyn) [inline src/controllers/Trips.js]: Hello @okabamac, kindly review the choice of return message. The message appears to require the line manager's physical presence

**Review by sebaztyn — CHANGES_REQUESTED** [review]

File: src/controllers/Trips.js (hunk 1 above)
Comment (okabamac) [reply src/controllers/Trips.js]: I think it's the same with yours. I just move Chidinma work

File: src/controllers/Trips.js [hunk 2]
```diff
@@ -177,6 +306,7 @@ class Trips {
           destinations: destinationList,
           departureDate,
           returnDate,
```
Comment (davidfrank96) [inline src/controllers/Trips.js:309]: Are u adding a returndate here?

**Review by davidfrank96 — APPROVED** [review]
pls take note if the comment is not relevant u can ignore but I think this is okay

File: src/controllers/Trips.js (hunk 2 above)
Comment (okabamac) [reply src/controllers/Trips.js:309]: Updated the GET Request to fetch the return date

**Review by mthamayor — CHANGES_REQUESTED** [review]
Hey mark. Nice work so far. However, I noticed in the request spec for multicity, the 'destination' property takes an array, while in `one-way` the  'destination` property takes an integer. I would suggest you follow the existing format of taking an array of integers. but in your case, only one value should be present in the array. Can you please look into it?

**Comment by okabamac** [comment]
> Hey mark. Nice work so far. However, I noticed in the request spec for multicity, the 'destination' property takes an array, while in `one-way` the 'destination` property takes an integer. I would suggest you follow the existing format of taking an array of integers. but in your case, only one value should be present in the array. Can you please look into it?

I have implemented this feedback. Can you please confirm?

**Review by mthamayor — CHANGES_REQUESTED** [review]
Hello. After signing up and receiving a token. I put the token in the Authorization header as `Bearer <token>`  and made a request. Then i get
```
{
    "status": "error",
    "message": "Cannot Find User With Email: undefined"
}
```
Can you please look into this?

**Comment by mthamayor** [comment]
> > Hey mark. Nice work so far. However, I noticed in the request spec for multicity, the 'destination' property takes an array, while in `one-way` the 'destination` property takes an integer. I would suggest you follow the existing format of taking an array of integers. but in your case, only one value should be present in the array. Can you please look into it?
> 
> I understand, but I don't think it is necessarily because I am still saving it as an array. And I was thinking it will be too much stress prompting the user to enter an array for a single value

This is why I asked the TTL about the format. I think it would do the team good if we have the same request for all the create trip requests endpoint

**Review by okafor-chidimma — APPROVED** [review]
tested locally and all tests are passing

**Comment by okabamac** [comment]
> > > Hey mark. Nice work so far. However, I noticed in the request spec for multicity, the 'destination' property takes an array, while in `one-way` the 'destination` property takes an integer. I would suggest you follow the existing format of taking an array of integers. but in your case, only one value should be present in the array. Can you please look into it?
> > 
> > 
> > I understand, but I don't think it is necessarily because I am still saving it as an array. And I was thinking it will be too much stress prompting the user to enter an array for a single value
> 
> This is why I asked the TTL about the format. I think it would do the team good if we have the same request for all the create trip requests endpoint

OK. Let me get clarifications. Thanks alot

**Comment by okabamac** [comment]
> Hello. After signing up and receiving a token. I put the token in the Authorization header as `Bearer <token>` and made a request. Then i get
> 
> ```
> {
>     "status": "error",
>     "message": "Cannot Find User With Email: undefined"
> }
> ```
> 
> Can you please look into this?

OK. It has been refactored by Chidinma and I sort of agree because only logged in user can make operations (user that has signed up and has verified his account). The token you get after signup doesnt attaches the email but when once you logged in that other token has email attached. Don't know if you're comfortable with it.

**Comment by okabamac** [comment]
> > > Hey mark. Nice work so far. However, I noticed in the request spec for multicity, the 'destination' property takes an array, while in `one-way` the 'destination` property takes an integer. I would suggest you follow the existing format of taking an array of integers. but in your case, only one value should be present in the array. Can you please look into it?
> > 
> > 
> > I understand, but I don't think it is necessarily because I am still saving it as an array. And I was thinking it will be too much stress prompting the user to enter an array for a single value
> 
> This is why I asked the TTL about the format. I think it would do the team good if we have the same request for all the create trip requests endpoint

I just implemented the feedback. Thanks a lot

**Review by sebaztyn — APPROVED** [review]

**Review by mthamayor — APPROVED** [review]

**Review by Stylll — APPROVED** [review]

**Pull request merged**